### DISCUSSION TITLE
Call the setup function

### DISCUSSION THREAD

**Discussion by erny0024** [ldelossa/nvim-ide#109] [Q&A]
What does it mean to call the setup function in step 2? 

Thanks.

**Comment by rabowhall** [comment]
I don't have an answer, but I wonder if you figured out how to call the setup function?   I have the same question!

**Comment by doctorfree** [comment]
Adding and configuring a Neovim/Vim plugin varies depending on which package manager you are using. I use Lazy and just copied the nvim-ide default configuration from the README and wrapped it inside Lazy's plugin initialization and configuration structure (see below). This is Lua so if you are adding this plugin in vimscript you will have to wrap the Lua code something like:

```
lua << EOF
setup goes here
EOF
```

It's hard to give specific instructions without knowing your setup. But maybe looking at my Lazy configuration for this plugin will help. This is the contents of the file `~/.config/nvim/lua/plugins/nvim-ide.lua`. The location of the file may vary, again depending on your package manager and setup. Generally, people have a `plugins` directory and it contains files with code that initializes and configures the plugins they want to use. Lazy automatically loads files in my plugins directory but some other package managers don't do that, you need to `require` the file from an `init.lua` or `init.vim` if you are using vimscript. Anyway, here is my Lazy setup:

```lua
return {
  {
    "ldelossa/nvim-ide",
    cmd = "Workspace",
    config = function()
      local bufferlist = require("ide.components.bufferlist")
      local explorer = require("ide.components.explorer")
      local outline = require("ide.components.outline")
      local callhierarchy = require("ide.components.callhierarchy")
      local timeline = require("ide.components.timeline")
      local terminal = require("ide.components.terminal")
      local terminalbrowser = require("ide.components.terminal.terminalbrowser")
      local changes = require("ide.components.changes")
      local commits = require("ide.components.commits")
      local branches = require("ide.components.branches")
      local bookmarks = require("ide.components.bookmarks")

      require("ide").setup({
        icon_set = "default",
        log_level = "info",
        components = {
          global_keymaps = {},
        },
        panels = {
          left = "explorer",
          right = "git",
        },
        panel_groups = {
          explorer = {
            outline.Name,
            bufferlist.Name,
            explorer.Name,
            bookmarks.Name,
            callhierarchy.Name,
            terminalbrowser.Name,
          },
          terminal = { terminal.Name },
          git = { changes.Name, commits.Name, timeline.Name, branches.Name },
        },
        workspaces = {
          auto_open = "left",
        },
        panel_sizes = {
          left = 30,
          right = 30,
          bottom = 15,
        },
      })
    end,
  },
}
```
